### ISSUE TITLE
 500 (Server Error) for failed uniqueness validations rather than a 400 (Bad Response)

### ISSUE THREAD

**Issue by HolySamosa** [balderdashy/sails#2498]
With sails-mongo, Sails still returns 500 (Server Error) for failed uniqueness validations rather than a 400 (Bad Response).  Not sure if this is true for other persistence implementations.

I've seen other related issues closed here.  I'd say this is is a major, as I imagine most applications need to enforce such a validation somewhere,  folks have to write kludges to work around the problem and the stack traces produced bloat and obscure the server log. 

Attention to this would be fantastic and appreciated!

**Comment by psi-4ward** [comment]
again: https://github.com/balderdashy/sails/issues/314#issuecomment-70387544
```javascript
      cb({
        code: 'E_UNIQUE',
...
        status: 400
      });

**Comment by CWyrtzen** [comment]
Is this still an issue?

**Comment by sailsbot** [comment]
Thanks for posting, @HolySamosa.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
